### PR TITLE
Remove constructor parameter from `Checks` class

### LINKED ISSUE #156: Remove constructor parameter from `Checks` class
### Is your enhancement related to a problem? Please describe.

Based on https://github.com/10up/plugin-check/issues/127#issuecomment-1499122587:
* The `$plugin_main_file` parameter should be removed from the `Checks` class constructor, since it is only used in the `run_checks()` method, which is conditionally called much later.
* The `Checks::context()` method should therefore also be removed.
* The `Checks::run_checks()` method should have its signature changed to `run_checks( Check_Context $context, array $checks )`, so that the `Check_Context` for the plugin to check only needs to be passed there.
* With that, we can also remove the `Abstract_Check_Runner::get_checks_instance()` method and underlying `Abstract_Check_Runner::$checks` property which has been confusing in terms of naming and now doesn't really serve a purpose anymore, since you can just write `new Checks()` and be done with it.
* Instead, we can add a `Abstract_Check_Runner::get_check_context()` method which returns `new Check_Context( WP_PLUGIN_DIR . '/' . $this->get_plugin_basename() )`.
* The `Abstract_Check_Runner::prepare()` method that uses the (now removed) `Checks::context()` method can then instead simply use the new `get_check_context()` method.

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
Closes #156 

In this PR, I have not removed the get_checks_instance method because the new Checks class creates a new instance for each request, and removing the method would cause the check to not work.

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: includes/Checker/Checks.php [hunk 1]
```diff
@@ -25,47 +25,25 @@ class Checks {
...
-	 * Returns the Check Context.
-	 *
-	 * @since n.e.x.t
-	 *
-	 * @return Check_Context The plugin context that is being checked.
-	 */
-	public function context() {
-		return $this->check_context;
-	}
+	public function __construct() { }
```
Comment (spacedmonkey) [inline includes/Checker/Checks.php]: This line can be removed.

Comment (mukeshpanchal27) [reply includes/Checker/Checks.php]: Thanks @spacedmonkey, Address in https://github.com/10up/plugin-check/pull/167/commits/efa0818f9985e2f8d1fde088d1ec576f63fc24be

File: includes/Checker/Abstract_Check_Runner.php [hunk 2]
```diff
@@ -346,4 +346,14 @@ public function get_plugin_basename() {
...
 		return $this->plugin_basename;
 	}
+
+	/** Gets the Check_Context for the plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Check_Context The check context for the plugin file.
+	 */
+	protected function get_check_context() {
```
Comment (felixarntz) [inline includes/Checker/Abstract_Check_Runner.php]: Better to make this private since it's not needed in child classes (related to #162).

**Review by felixarntz — COMMENTED** [review]
@mukeshpanchal27 Almost LGTM, just one small thing to enhance. Normally not a blocker, but since we're fixing this via #162, we should also stick to that pattern here, to use the lowest visibility possible for each method.

File: tests/Checker/Checks_Tests.php [hunk 3]
```diff
@@ -15,7 +16,8 @@ class Checks_Tests extends WP_UnitTestCase {
 	public function set_up() {
 		parent::set_up();
 
-		$this->checks = new Checks( 'test-plugin/test-plugin.php' );
+		$this->checks  = new Checks( 'test-plugin/test-plugin.php' );
```
Comment (jjgrainger) [inline tests/Checker/Checks_Tests.php]: @mukeshpanchal27 we can remove the constructor parameter here

File: includes/Checker/Abstract_Check_Runner.php [hunk 4]
```diff
@@ -311,7 +311,7 @@ protected function get_checks_instance() {
 		}
 
 		$plugin_basename = $this->get_plugin_basename();
```
Comment (jjgrainger) [inline includes/Checker/Abstract_Check_Runner.php]: @mukeshpanchal27 we can remove this line also as we no longer need the plugin basename here

**Review by jjgrainger — CHANGES_REQUESTED** [review]
Thanks @mukeshpanchal27 looking good, just a few minor changes to be made

**Review by felixarntz — APPROVED** [review]
@mukeshpanchal27 Approved from my end, pending the changes from @jjgrainger.

File: tests/Checker/Checks_Tests.php (hunk 3 above)
Comment (mukeshpanchal27) [reply tests/Checker/Checks_Tests.php]: Good catch. Addressed in https://github.com/10up/plugin-check/pull/167/commits/ab432f0c9e691f45e9800dc7014da8f948b09942

**Review by jjgrainger — APPROVED** [review]

**Pull request merged**